tests/sycl/group_functions/group_functions.hpp

```diff
@@ -261,6 +261,19 @@ void check_binary_reduce(std::vector<T> buffer, size_t local_size, size_t global
   }
 }
 
+#if defined(HIPSYCL_PLATFORM_CUDA) || defined(HIPSYCL_PLATFORM_HIP)
+inline size_t compute_subgroup_size(int i, size_t local_size) {
+  size_t warp_start = static_cast<int>((i % local_size) / warpSize) * warpSize;
+  return (warpSize < (local_size - warp_start)) ? warpSize : (local_size - warp_start);
+}
+
+inline bool last_in_subgroup(int i, size_t local_size) {
+  size_t index_in_work_group = i % local_size;
+  size_t warp_size = compute_subgroup_size(i, local_size);
+  return ((index_in_work_group % warpSize) + 1  == warp_size);
+}
```

Comment:
Those functions are only used in the host side for the tests, right? Then please do not use `warpSize` as it doesn't work in explicit multipass.
The proper way is to query the subgroup size of a SYCL device using the `device::get_info` API :-)

Reply:
I have an idea that might make the test more readable and removes the need for these functions. So hopefully they aren't needed anymore :)

Should I replace all occurrences of `warpSize` in the tests (only used for GPU-tests anyways)?